### PR TITLE
refactor(query): improve to_string(number) function

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Closes #issue

### CODE REVIEW

**Comment by sundy-li** [comment]
```
-- before
MySQL [default]> select count() from numbers_mt(1000000000) where ignore( to_string(number) );
+---------+
| count() |
+---------+
|       0 |
+---------+
1 row in set (1.320 sec)

-- after
MySQL [default]> select count() from numbers_mt(1000000000) where ignore( to_string(number) );
+---------+
| count() |
+---------+
|       0 |
+---------+
1 row in set (0.508 sec)
```

File: src/query/expression/src/types/number.rs [hunk 1]
```diff
@@ -707,6 +707,8 @@ pub trait Number:
     + Send
     + 'static
 {
+    type Native;
```
Comment (Xuanwo) [inline src/query/expression/src/types/number.rs:710]: Cool!

**Review by BohuTANG — APPROVED** [review]

**Comment by BohuTANG** [comment]
Unit test failed related to this PR :)

File: src/query/functions/src/scalars/arithmetic.rs [hunk 2]
```diff
@@ -393,25 +397,73 @@ pub fn register(registry: &mut FunctionRegistry) {
...
                         FunctionProperty::default(),
                         |_| FunctionDomain::Full,
-                        vectorize_with_builder_1_arg::<NumberType<NUM_TYPE>, StringType>(
-                            |val, output, _| {
-                                output.write_row(|data| write!(data, "{val}").unwrap());
-                                Ok(())
-                            },
-                        ),
+                        |from, _| match from {
+                            ValueRef::Scalar(s) => Ok(Value::Scalar(format!("{}", s).into_bytes())),
```
Comment (andylokandy) [inline src/query/functions/src/scalars/arithmetic.rs:401]:
```suggestion
                            ValueRef::Scalar(s) => Ok(Value::Scalar(s.to_string().into_bytes())),
```

File: src/query/functions/src/scalars/arithmetic.rs [hunk 3]
```diff
@@ -393,25 +397,73 @@ pub fn register(registry: &mut FunctionRegistry) {
...
+
+                            type Native = <NUM_TYPE as Number>::Native;
+                            let mut offset: usize = 0;
+                            unsafe {
+                                for x in from.iter() {
+                                    values.reserve(offset + Native::FORMATTED_SIZE_DECIMAL);
+                                    let bytes = std::slice::from_raw_parts_mut(
+                                        values.as_mut_ptr().add(offset),
+                                        values.capacity() - offset,
+                                    );
```
Comment (andylokandy) [inline src/query/functions/src/scalars/arithmetic.rs:450]: Could it be written to
```rust
let bytes = &mut builder.data[offset..];
```

**Comment by sundy-li** [comment]
continue in #9501

**Pull request closed without merging**